amr-wind/equation_systems/vof/volume_fractions.H

```diff
@@ -1,34 +1,249 @@
+         volfrac(i, j + 1, k);
+    m[0][1] = 0.5 * (m1 - m2);
+
+    m1 = volfrac(i - 1, j, k - 1) + volfrac(i + 1, j, k - 1) +
+         volfrac(i, j, k - 1);
+    m2 = volfrac(i - 1, j, k + 1) + volfrac(i + 1, j, k + 1) +
+         volfrac(i, j, k + 1);
+    m[0][2] = 0.5 * (m1 - m2);
+
+    // write the plane as: sgn(my) Y =  mx X +  mz Z + alpha,
+    //                          m11 Y = m10 X + m12 Z + alpha.
+
+    m1 = volfrac(i - 1, j - 1, k) + volfrac(i - 1, j + 1, k) +
+         volfrac(i - 1, j, k);
+    m2 = volfrac(i + 1, j - 1, k) + volfrac(i + 1, j + 1, k) +
+         volfrac(i + 1, j, k);
+    m[1][0] = 0.5 * (m1 - m2);
+
+    m1 = volfrac(i, j - 1, k - 1) + volfrac(i, j - 1, k + 1) +
+         volfrac(i + 1, j - 1, k) + volfrac(i - 1, j - 1, k) +
+         volfrac(i, j - 1, k);
+    m2 = volfrac(i, j + 1, k - 1) + volfrac(i, j + 1, k + 1) +
+         volfrac(i + 1, j + 1, k) + volfrac(i - 1, j + 1, k) +
+         volfrac(i, j + 1, k);
+
+    if (m1 > m2) {
+        m[1][1] = 1.;
+    } else {
+        m[1][1] = -1.;
+    }
+
+    m1 = volfrac(i, j - 1, k - 1) + volfrac(i, j, k - 1) +
+         volfrac(i, j + 1, k - 1);
+    m2 = volfrac(i, j - 1, k + 1) + volfrac(i, j, k + 1) +
+         volfrac(i, j + 1, k + 1);
+    m[1][2] = 0.5 * (m1 - m2);
+
+    // write the plane as: sgn(mz) Z =  mx X +  my Y + alpha
+    //                          m22 Z = m20 X + m21 Y + alpha
+
+    m1 = volfrac(i - 1, j, k - 1) + volfrac(i - 1, j, k + 1) +
+         volfrac(i - 1, j, k);
+    m2 = volfrac(i + 1, j, k - 1) + volfrac(j + 1, j, k + 1) +
+         volfrac(i + 1, j, k);
+    m[2][0] = 0.5 * (m1 - m2);
+
+    m1 = volfrac(i, j - 1, k - 1) + volfrac(i, j - 1, k + 1) +
+         volfrac(i, j - 1, k);
+    m2 = volfrac(i, j + 1, k - 1) + volfrac(i, j + 1, k + 1) +
+         volfrac(i, j + 1, k);
+    m[2][1] = 0.5 * (m1 - m2);
+
+    m1 = volfrac(i - 1, j, k - 1) + volfrac(i + 1, j, k - 1) +
+         volfrac(i, j - 1, k - 1) + volfrac(i, j + 1, k - 1) +
+         volfrac(i, j, k - 1);
+    m2 = volfrac(i - 1, j, k + 1) + volfrac(i + 1, j, k + 1) +
+         volfrac(i, j - 1, k + 1) + volfrac(i, j + 1, k + 1) +
+         volfrac(i, j, k + 1);
+
+    if (m1 > m2) {
+        m[2][2] = 1.;
+    } else {
+        m[2][2] = -1.;
+    }
+
+    // normalize each set (mx,my,mz): |mx|+|my|+|mz| = 1
+    amrex::Real t0, t1, t2;
+
+    t0 = amrex::Math::abs(m[0][0]) + amrex::Math::abs(m[0][1]) +
+         amrex::Math::abs(m[0][2]);
+    m[0][0] = m[0][0] / t0;
+    m[0][1] = m[0][1] / t0;
+    m[0][2] = m[0][2] / t0;
+
+    t0 = amrex::Math::abs(m[1][0]) + amrex::Math::abs(m[1][1]) +
+         amrex::Math::abs(m[1][2]);
+    m[1][0] = m[1][0] / t0;
+    m[1][1] = m[1][1] / t0;
+    m[1][2] = m[1][2] / t0;
+
+    t0 = amrex::Math::abs(m[2][0]) + amrex::Math::abs(m[2][1]) +
+         amrex::Math::abs(m[2][2]);
+    m[2][0] = m[2][0] / t0;
+    m[2][1] = m[2][1] / t0;
+    m[2][2] = m[2][2] / t0;
+
```

Comment:
Replace with `vs::Vector` and use 

```C++
for (int i=0; i < AMREX_SPACEDIM; ++i) m[i].normalize();
```

Reply:
Not sure what vs::Vector is it different to the std::vector ?